src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerPendingHearingOutcome.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.divorce.caseworker.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingDwpResponse;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.GeneralConsiderationComplete;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.PendingHearingOutcome;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class CaseworkerPendingHearingOutcome implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_PENDING_HEARING_OUTCOME = "caseworker-pending-hearing-outcome";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_PENDING_HEARING_OUTCOME)
+            .forStates(GeneralConsiderationComplete, AwaitingDwpResponse)
+            .name("Pending hearing outcome")
+            .description("Pending hearing outcome")
+            .showEventNotes()
+            .showSummary()
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
+            .grantHistoryOnly(SOLICITOR));
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+        log.info("Caseworker pending hearing outcome about to submit callback invoked");
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .state(PendingHearingOutcome)
```

Comment:
if we just need to transition the state we can do that in forStateTransition in even config ?

Reply:
yes sorry forgot about the EnumSet.of, thank you!